src/mango.ts

```diff
@@ -0,0 +1,491 @@
+    scalper.connection,
+    scalper.symbol,
+    scalper.owner,
+    spotMarket,
+  );
+
+  // Get Total Delta Position to hedge
+  let hedgeDeltaTotal = IS_DEV
+    ? 0.1
+    : mangoPerpDelta + dipTotalDelta + spotDelta + mangoSpotDelta + scalper.deltaOffset;
+
+  // Check whether we need to hedge.
+  const dipTotalGamma = getDIPGamma(dipProduct, fairValue, scalper.symbol);
+  const stdDevSpread = (scalper.impliedVol / Math.sqrt(SEC_PER_YEAR
+    / SCALPER_WINDOW_SEC)) * scalper.zScore;
+  const deltaThreshold = Math.max(
+    dipTotalGamma * stdDevSpread * fairValue * GAMMA_THRESHOLD,
+    scalper.minSize,
+  );
+  if (Math.abs(hedgeDeltaTotal) < (deltaThreshold)) {
+    console.log(scalper.symbol, 'Delta Netural', hedgeDeltaTotal, '<', deltaThreshold);
+    return;
+  }
+
+  // TODO: Add slippage delta adjustment
+  const slippageTolerance = Math.min(stdDevSpread / 2, slippageMax.get(scalper.symbol));
+  let hedgePrice = hedgeDeltaTotal < 0
+    ? fairValue * (1 + slippageTolerance)
+    : fairValue * (1 - slippageTolerance);
+  const slippageDIPDelta = getDIPDelta(dipProduct, hedgePrice, scalper.symbol);
+  const dipDeltaDiff = slippageDIPDelta - dipTotalDelta;
+  hedgeDeltaTotal += dipDeltaDiff;
+  console.log(scalper.symbol, 'Adjust Slippage Delta by', dipDeltaDiff, 'to', -hedgeDeltaTotal);
+  const hedgeSide = hedgeDeltaTotal < 0 ? PerpOrderSide.bid : PerpOrderSide.ask;
+  const hedgeSideText = hedgeSide === PerpOrderSide.bid ? 'Buy' : 'Sell';
+
+  const notionalThreshold = deltaThreshold * fairValue;
+  const notionalAmount = -hedgeDeltaTotal * fairValue;
+  if ((notionalAmount < notionalThreshold && hedgeSideText === 'Buy')
+       || (notionalAmount > notionalThreshold && hedgeSideText === 'Sell')) {
+    console.log(scalper.symbol, 'Delta Netural: Slippage', deltaThreshold);
+    return;
+  }
+  console.log(scalper.symbol, 'Outside delta threshold:', Math.abs(hedgeDeltaTotal), 'vs.', deltaThreshold);
+
+  // Determine spot or perp order based on funding rate.
+  const bidSide = await perpMarket.loadBids(mangoClient);
+  const askSide = await perpMarket.loadAsks(mangoClient);
+  const fundingRate = (24 * 365)
+    * perpMarket.getCurrentFundingRate(bidSide, askSide);
+  const buySpot = fundingRate > PERP_FUNDING_RATE_THRESHOLD;
+  const sellSpot = -fundingRate < PERP_FUNDING_RATE_THRESHOLD;
+  let hedgeProduct = getMangoHedgeProduct(hedgeSide, buySpot, sellSpot);
+  // No OpenBook BTC market exists still
+  if (scalper.symbol === 'BTC') {
+    hedgeProduct = HedgeProduct.Perp;
+  }
+
+  console.log(
+    `${scalper.symbol} Target Delta Hedge: ${hedgeSideText} ${hedgeProduct} ${-hedgeDeltaTotal} \
+      DIP Δ: ${dipTotalDelta} Mango Perp Δ: ${mangoPerpDelta} Mango Spot Δ: ${mangoSpotDelta} \
+      Spot Δ: ${spotDelta} Offset Δ ${scalper.deltaOffset} Fair Value: ${fairValue}`,
+  );
+
+  // Determine what price to use for hedging depending on allowable slippage.
+  hedgePrice = roundPriceToTickSize(hedgeDeltaTotal < 0
```

Comment:
we use hedgeDeltaTotal < 0 a lot, put it into a named variable explaining what that means

Reply:
IS_BUYSIDE added